dronegen/tag.go

```diff
@@ -0,0 +1,425 @@
+		environment["FIPS"] = value{raw: "yes"}
+		environment["RUNTIME"] = value{raw: "fips"}
+		makeCommand = fmt.Sprintf("make -C e %s", packageType)
+	} else {
+		environment["OSS_TARBALL_PATH"] = value{raw: "/go/artifacts"}
+	}
+
+	tagVolumes := []volume{
+		volumeDocker,
+	}
+	tagVolumeRefs := []volumeRef{
+		volumeRefDocker,
+	}
+	if packageType == rpmPackage {
+		environment["GNUPG_DIR"] = value{raw: "/tmpfs/gnupg"}
+		environment["GPG_RPM_SIGNING_ARCHIVE"] = value{fromSecret: "GPG_RPM_SIGNING_ARCHIVE"}
+		packageBuildCommands = append(packageBuildCommands,
+			`mkdir -m0700 $GNUPG_DIR`,
+			`echo "$GPG_RPM_SIGNING_ARCHIVE" | base64 -d | tar -xzf - -C $GNUPG_DIR`,
+			`chown -R root:root $GNUPG_DIR`,
+			makeCommand,
+			`rm -rf $GNUPG_DIR`,
+		)
+		tagVolumes = []volume{
+			volumeDocker,
+			volumeTmpfs,
+		}
+		tagVolumeRefs = []volumeRef{
+			volumeRefDocker,
+			volumeRefTmpfs,
+		}
+
+	} else if packageType == debPackage {
+		packageBuildCommands = append(packageBuildCommands,
+			makeCommand,
+		)
+	}
+
+	pipelineName := fmt.Sprintf("%s-%s", dependentPipeline, packageType)
+
+	p := newKubePipeline(pipelineName)
+	p.Trigger = triggerTag
+	p.DependsOn = []string{dependentPipeline}
+	p.Workspace = workspace{Path: "/go"}
+	p.Volumes = tagVolumes
+	p.Services = []service{
+		{
+			Name:    "Start Docker",
+			Image:   "docker:dind",
+			Volumes: tagVolumeRefs,
+		},
+	}
+	p.Steps = []step{
+		{
+			Name:  "Check out code",
+			Image: "docker:git",
+			Environment: map[string]value{
+				"GITHUB_PRIVATE_KEY": value{fromSecret: "GITHUB_PRIVATE_KEY"},
+			},
+			Commands: tagCheckoutCommands(params.fips),
+		},
+		{
+			Name:  "Download built tarball artifacts from S3",
+			Image: "amazon/aws-cli",
+			Environment: map[string]value{
+				"AWS_REGION":            value{raw: "us-west-2"},
+				"AWS_S3_BUCKET":         value{fromSecret: "AWS_S3_BUCKET"},
+				"AWS_ACCESS_KEY_ID":     value{fromSecret: "AWS_ACCESS_KEY_ID"},
+				"AWS_SECRET_ACCESS_KEY": value{fromSecret: "AWS_SECRET_ACCESS_KEY"},
+			},
+			Commands: tagDownloadArtifactCommands(params),
+		},
+		{
+			Name:        fmt.Sprintf("Build %s artifacts", strings.ToUpper(packageType)),
+			Image:       "docker",
+			Environment: environment,
+			Volumes:     tagVolumeRefs,
+			Commands:    packageBuildCommands,
+		},
+		{
+			Name:     fmt.Sprintf("Copy %s artifacts", strings.ToUpper(packageType)),
+			Image:    "docker",
+			Commands: tagCopyPackageArtifactCommands(params, packageType),
+		},
+		{
+			Name:  "Upload to S3",
+			Image: "plugins/s3",
+			Settings: map[string]value{
+				"bucket":       value{fromSecret: "AWS_S3_BUCKET"},
+				"access_key":   value{fromSecret: "AWS_ACCESS_KEY_ID"},
+				"secret_key":   value{fromSecret: "AWS_SECRET_ACCESS_KEY"},
+				"region":       value{raw: "us-west-2"},
+				"source":       value{raw: "/go/artifacts/*"},
+				"target":       value{raw: "teleport/tag/${DRONE_TAG##v}"},
+				"strip_prefix": value{raw: "/go/artifacts/"},
+			},
+		},
```

Comment:
optional: extract this into `func uploadToS3Step(s s3Settings) step`, using
```
type s3Settings struct {
  region string
  source string
  target string
  stripPrefix string
}
```
and inject the secret values automatically